opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -288,20 +288,30 @@ private void reduceEvent(OnmsEvent persistedEvent, OnmsAlarm alarm, Event event)
             }
         }
 
-        Set<OnmsAlarm> relatedAlarms = getRelatedAlarms(event.getParmCollection());
-        if (relatedAlarms != null && !relatedAlarms.isEmpty()) {
-            // alarm.relatedAlarms becomes the union of any existing alarms and any in the event.
-            for (OnmsAlarm related : relatedAlarms) {
-                if (!formingCyclicGraph(alarm, related)) {
-                    alarm.addRelatedAlarm(related);
-                } else {
-                    LOG.warn("Alarm with id '{}' , reductionKey '{}' is not added as related alarm for id '{}' as it is forming cyclic graph ", related.getId(), related.getReductionKey(), alarm.getId());
-                }
-            }
-        }
+        updateRelatedAlarms(alarm, event);
 
         persistedEvent.setAlarm(alarm);
     }
+    
+    private void updateRelatedAlarms(OnmsAlarm alarm, Event event) {
+        // Clear the existing related alarms that may be known for this alarm so that we treat the event as an
```

Comment:
Can we add a test for this behavior?

Reply:
Yep, there was actually already some cases that checked for the old behaviour so I updated them to expect the new behaviour.